experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -470,23 +464,146 @@ export interface Sequence<TTypes extends AllowedTypes> extends TreeField {
+	findLast<T extends UnboxNodeUnion<TTypes>>(
+		predicate: (element: UnboxNodeUnion<TTypes>, index: number, sequence: this) => element is T,
+	): T | undefined;
+
+	/**
+	 * Returns the index of the first element in the sequence where `predicate` is true, otherwise -1.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	findIndex(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): number;
+
+	/**
+	 * Returns the index of the last element in the sequence where `predicate` is true, otherwise -1.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	findLastIndex(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): number;
 
 	/**
 	 * Calls the provided callback function on each child of this sequence, and returns an array that contains the results.
 	 * @param callbackfn - A function that accepts the child, its index, and this field.
 	 */
-	mapBoxed<U>(callbackfn: (value: TypedNodeUnion<TTypes>, index: number, array: this) => U): U[];
+	map<U>(callbackfn: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => U): U[];
 
+	/**
+	 * Calls the specified callback function for all the elements in the sequence.
+	 * The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.
+	 * @param callbackfn - A function that accepts up to four arguments.
+	 * The reduce method calls the callbackfn function one time for each element in the array.
+	 * @param initialValue - If present, it is passed to the first call to the callbackfn.
+	 * Otherwise, the first element in the sequence will be used as the initial accumulator value and skipped as currentValue.
+	 */
+	reduce(
+		callbackfn: (
+			previousValue: UnboxNodeUnion<TTypes>,
+			currentValue: UnboxNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => UnboxNodeUnion<TTypes>,
+		initialValue?: UnboxNodeUnion<TTypes>,
+	): UnboxNodeUnion<TTypes>;
+	reduce<U>(
+		callbackfn: (
+			previousValue: U,
+			currentValue: UnboxNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => UnboxNodeUnion<TTypes>,
+		initialValue: U,
+	): U;
+
+	/**
+	 * Returns the elements of the sequence that meet the condition specified in a callback function.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	filter(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): UnboxNodeUnion<TTypes>[];
+
+	/**
+	 * Returns true if any element of the sequence satisfies the given predicate.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	some(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): boolean;
+
+	/**
+	 * Returns true if every element of the sequence satisfies the given predicate.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	every(
```

Comment:
I think reduce or fold are better patterns which are more general.

Reply:
Yeah.  It's a pure subset in terms of functionality...  although perhaps it improves readability for this common pattern.  Either way, there's little upside in omitting it.